fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/fragments/event/creation/ElectionSetupFragment.java

```diff
@@ -0,0 +1,202 @@
+package com.github.dedis.student20_pop.detail.fragments.event.creation;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Spinner;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import com.github.dedis.student20_pop.databinding.FragmentSetupElectionEventBinding;
+import com.github.dedis.student20_pop.detail.LaoDetailActivity;
+import com.github.dedis.student20_pop.detail.LaoDetailViewModel;
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+public class ElectionSetupFragment extends AbstractEventCreationFragment implements AdapterView.OnItemSelectedListener {
+
+    public static final String TAG = ElectionSetupFragment.class.getSimpleName();
+
+    private FragmentSetupElectionEventBinding mSetupElectionFragBinding;
+
+    //mandatory fields for submitting
+    private EditText electionNameText;
+    private EditText electionQuestionText;
+    private Button submitButton;
+
+    private enum votingMethods {Plurality}
```

Comment:
Is there a reason for making this private? I think this might cause an issue if you were to test something based on `VotingMethods`

Reply:
Since I only use this enum in this fragment, I thought it would be best to make it private, but I agree that it's logical to make it public. And yes, I will change the enum to follow the convention, my bad !